extensions/amp-consent/0.1/amp-consent.js

```diff
@@ -524,6 +531,51 @@ export class AmpConsent extends AMP.BaseElement {
     });
   }
 
+  /**
+   * Sync consent string and consent state value to local storage.
+   */
+  syncConsentStringAndStateValue_() {
+    this.getConsentRemote_().then(response => {
+      // Decision from promptUI takes precedence over response
+      if (response && !this.consentStateChangedViaPromptUI_) {
+        this.consentStateManager_
+          .getLastConsentInstanceInfo()
+          .then(storedInfo => {
+            const consentRequired = response['consentRequired'];
+            const consentState = convertValueToState(
+              response['consentStateValue']
+            );
+            // Local storage empty, consent required, and state is not unknown,
+            // then use the state given.
+            if (
+              !!consentRequired &&
+              consentState !== CONSENT_ITEM_STATE.UNKNOWN
```

Comment:
we should clear cache if it's `"unknown"`. we don't touch cache if it's `null`

Reply:
> we should clear cache if it's "unknown". 

sg
>  we don't touch cache if it's null 

this is handled in updateCacheIfNotNull() and also in consentStateManagaer_.update(), where it checks that the new consentInstance is different, before actually updating.